### PR TITLE
Added username lookup

### PR DESCRIPTION
**Pull request by J3fftw1**
Deprecated the old lookup
added a new one which has caching 

Would fix ratelimiting issues

### CODE REVIEW

**Comment by md5sha256** [comment]
I'm not entirely sure how the Http API in the Java works but I'm pretty sure you're not supposed to create a new Http client every run. I'm gonna check, but the last time I made a new one every time I did hit ratelimiting issues

**Comment by J3fftw1** [comment]
> I'm not entirely sure how the Http API in the Java works but I'm pretty sure you're not supposed to create a new Http client every run. I'm gonna check but the last time I made a new one every time I did hit ratelimiting issues

im not exactly sure how it works either i just straight up copied from the javadocs which said this.
also i doubt we would get ratelimted by Cloudflare for this.
Would be interesting to see but if thats the case we can just scream at @WalshyDev

**Comment by WalshyDev** [comment]
Making a new HTTP client wouldn't create rate-limiting by itself, all we'd probably do is prevent re-use of connections (which is also not something we want to do).

So yes, please move that up and keep re-using the same client. However, can't see that alone causing rate-limiting. The API we're using here also does no rate-limiting.

**Comment by md5sha256** [comment]
> Making a new HTTP client wouldn't create rate-limiting by itself, all we'd probably do is prevent re-use of connections (which is also not something we want to do).
> 
> So yes, please move that up and keep re-using the same client. However, can't see that alone causing rate-limiting. The API we're using here also does no rate-limiting.

Ah, I may have mistook a reused connection for a ratelimit as all I got was a timeout exception last time I tried. But anyhow, re-sounds good.

File: dough-skins/src/main/java/io/github/bakedlibs/dough/skins/UUIDLookup.java [hunk 1]
```diff
@@ -28,9 +32,58 @@ public class UUIDLookup {
...
+                .timeout(Duration.ofMinutes(2))
+                .header("user-agent", "Mozilla/5.0 Dough (+https://github.com/baked-libs/dough)")
+                .GET()
+                .build();
+
+        return HTTP_CLIENT.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+                .thenApply(HttpResponse::body)
+                .thenApply(s -> JSON_PARSER.parse(s).getAsJsonObject())
+                .thenApply(jsonObject -> {
+                    if (jsonObject.get("code").getAsString().equals("player.found")) {
```
Comment (WalshyDev) [inline dough-skins/src/main/java/io/github/bakedlibs/dough/skins/UUIDLookup.java:68]:
```suggestion
                    if (jsonObject.get("success").getAsBoolean()) {
```

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**